app/views/lookup_keys/_value.html.erb

```diff
@@ -9,7 +9,7 @@
     <%= text_field_tag '', nil, :class => 'matcher_value' %>
   </td>
   <td>
-    <%= hidden_value_field(f, :value, f.object.value, f.object.use_puppet_default, :'data-property' => 'value') %>
+    <%= hidden_value_field(f, :value, f.object.value_before_type_cast, f.object.use_puppet_default, :'data-property' => 'value') %>
```

Comment:
Right, so this is just intending to fix the case when loading the form and a hidden value is ticked.

I quite agree about using different tasks - the combining of various smart class variable bugs in this PR has made it difficult to review, as the contents keep changing.  This seems more related to #6007 rather than #11862.